.github/workflows/test-java.yaml

```
@@ -15,11 +15,47 @@ jobs:
   test:
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest]
-    runs-on: ${{matrix.os}}
+        # start with the most popular ones
+        include:
+          - os: macos-latest
+            target: x86_64-apple-darwin
+            jdk: 8
+          - os: macos-latest
+            target: aarch64-apple-darwin
+            # https://adoptium.net/supported-platforms/
+            jdk: 11
+          - os: ubuntu-latest
+            target: x86_64-unknown-linux-musl
+            jdk: 8
+          - os: windows-latest
+            target: x86_64-pc-windows-msvc
+            jdk: 8
+    runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
+      - name: Install JDK 8 and Maven
+        uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: ${{ matrix.jdk }}
+          cache: "maven"
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          target: ${{ matrix.target }}
+          override: true
+      - name: Build prqlc
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --manifest-path prql-compiler/prqlc/Cargo.toml
```

slightly more idomatic

```suggestion
          args: -p prqlc
```